src/components/Asset/Edit/EditMetadata.tsx

```diff
@@ -36,10 +37,27 @@ export default function Edit({
   const { accountId, web3 } = useWeb3()
   const newAbortController = useAbortController()
   const [success, setSuccess] = useState<string>()
+  const [paymentCollector, setPaymentCollector] = useState<string>()
   const [error, setError] = useState<string>()
   const isComputeType = asset?.services[0]?.type === 'compute'
   const hasFeedback = error || success
 
+  useEffect(() => {
+    async function getInitialPaymentCollector() {
+      let paymentCollector
+      try {
+        const datatoken = new Datatoken(web3)
+        paymentCollector = await datatoken.getPaymentCollector(
+          asset?.datatokens[0].address
+        )
+        setPaymentCollector(paymentCollector)
```

Comment:
```suggestion
        setPaymentCollector(await datatoken.getPaymentCollector(
          asset?.datatokens[0].address
        ))
```
can't we do this inline this way ?

Reply:
Yep, done